src/Driver/SolverTypes/IMEXSolverType.jl

```diff
@@ -0,0 +1,160 @@
+    Default: `ManyColumnLU`
+- `implicit_solver_adjustable` (Bool): A flag identifying whether
+    or not the `implicit_solver` can be updated as the time-step
+    size changes.
+    Default: `false`
+- `solver_method` (Function): Function defining the particular additive
+    Runge-Kutta method to be used for the IMEX method.
+    Default: `ARK2GiraldoKellyConstantinescu`
+- `solver_storage_variant` (Type): Storage type for the additive
+    Runge-Kutta method.
+    Default: `LowStorageVariant()`
+
+### References
+    @article{giraldo2013implicit,
+      title={Implicit-explicit formulations of a three-dimensional
+             nonhydrostatic unified model of the atmosphere ({NUMA})},
+      author={Giraldo, Francis X and Kelly, James F and Constantinescu, Emil M},
+      journal={SIAM Journal on Scientific Computing},
+      volume={35},
+      number={5},
+      pages={B1162--B1194},
+      year={2013},
+      publisher={SIAM}
+    }
+"""
+struct IMEXSolverType{DS, ST} <: AbstractSolverType
+    # The type of discrete splitting to apply to the right-hand side
+    splitting_type::DS
+    # Linear model describing fast dynamics
+    linear_model::Type
+    # Choice of implicit solver
+    implicit_solver::Type
+    # Can the implicit solver be updated with changing dt?
+    implicit_solver_adjustable::Bool
+    # Function for the IMEX method
+    solver_method::Function
+    # Storage type for the ARK scheme
+    solver_storage_variant::ST
+
+    function IMEXSolverType(;
+        splitting_type = HEVISplitting(),
+        linear_model = AtmosAcousticGravityLinearModel,
+        implicit_solver = ManyColumnLU,
+        implicit_solver_adjustable = false,
+        solver_method = ARK2GiraldoKellyConstantinescu,
+        solver_storage_variant = LowStorageVariant(),
+    )
+
+        DS = typeof(splitting_type)
+        ST = typeof(solver_storage_variant)
+
+        return new{DS, ST}(
+            splitting_type,
+            linear_model,
+            implicit_solver,
+            implicit_solver_adjustable,
+            solver_method,
+            solver_storage_variant,
+        )
+    end
+end
+
+"""
+# Description
+    solversetup(
+        ode_solver::IMEXSolverType{DS},
+        dg,
+        Q,
+        dt,
+        t0,
+        diffusion_direction,
+    ) where {DS <: HEVISplitting}
+
+Creates an ODE solver using a HEVI-type time-integration
+scheme. All horizontal acoustic waves are treated explicitly,
+while the 1-D vertical problem is treated implicitly. All
+other dynamics (advection, diffusion) is treated explicitly
+in the additive Runge-Kutta method.
+
+# Comments:
+Currently, the only HEVI-type splitting ClimateMachine can currently
+do only involves splitting the acoustic processes; it is not currently
+possible to perform more fine-grained separation of tendencies
+(for example, including vertical advection or diffusion in the 1-D implicit problem)
+"""
+function solversetup(
+    ode_solver::IMEXSolverType{DS},
```

Comment:
Here and in other places I think you can just do `IMEXSolverType{HEVISplitting}` and get rid of the `where` clause since the splitting type is concrete.